ethers-core/src/types/chain.rs

```diff
@@ -71,6 +57,32 @@ impl From<Chain> for u64 {
     }
 }
 
+impl TryFrom<u64> for Chain {
+    type Error = TryFromUnknownChain;
```

Comment:
would be ok to just make this a `()`,
which makes it essentially an Option

Reply:
I'm not sure I understand, do you mind elaborating?

Reply:
the error type is not really useful here since it's obvious, 
so we could simply make this `type Error = ();`